Content.Client/Disposal/Systems/DisposalUnitSystem.cs

```diff
@@ -101,12 +101,16 @@ private void UpdateState(EntityUid uid, SharedDisposalUnitComponent unit, Sprite
         sprite.LayerSetVisible(DisposalUnitVisualLayers.Base, state == VisualState.Anchored);
         sprite.LayerSetVisible(DisposalUnitVisualLayers.BaseFlush, state is VisualState.Flushing or VisualState.Charging);
 
+        sprite.LayerMapTryGet(DisposalUnitVisualLayers.BaseCharging, out var chargingLayer);
+        var chargingState = sprite.LayerGetState(chargingLayer);
```

Comment:
Have you tested what happens if this fails to get the layer/state, like if it's not defined in the prototype?

Reply:
My bad, just assumed that this would be fine. Anyways if it's not defined then it will be set to StateId.Invalid which is obviously incorrect. I'll fix it one moment.